lib/smart_todo/slack_client.rb

```diff
@@ -63,25 +64,27 @@ def post_message(channel, text)
 
     private
 
-    # @param method [Symbol]
-    # @param endpoint [String]
-    # @param data [String] JSON encoded data when making a POST request
-    # @param headers [Hash]
+    # @param request [Net::HTTPRequest]
+    # @param max_attempts [Integer] - the maximum number of attempts to make
+    # @param min_sleep [Integer] - the minimum sleep time in seconds in case of rate limiting
     #
     # @raise [Net::HTTPError] in case the request to Slack failed
     # @raise [SlackClient::Error] in case Slack returns a { ok: false } in the body
-    def dispatch(request, max_attempts = 5)
+    def dispatch(request, max_attempts = 5, min_sleep = 30)
       response = @client.request(request)
       attempts = 1
 
       while response.is_a?(Net::HTTPTooManyRequests) && attempts < max_attempts
-        sleep([Integer(response["Retry-After"]), 600].min)
+        sleep_time = Integer(response["Retry-After"]).clamp(min_sleep, 600)
```

Comment:
```suggestion
        sleep_time = Integer(response["Retry-After"]).clamp(min_sleep, 300)
```
10 minutes seems like a lot. Though, I don't know what the typical timeout is like.